.github/workflows/contrib-openai.yml

```diff
@@ -261,76 +262,112 @@ jobs:
-        - name: Install packages and dependencies
-          run: |
-            docker --version
-            python -m pip install --upgrade pip wheel
-            pip install -e .[lmm]
-            python -c "import autogen"
-            pip install coverage pytest
-        - name: Coverage
-          env:
-            OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
-          run: |
-            coverage run -a -m pytest test/agentchat/contrib/capabilities/test_image_generation_capability.py
-            coverage xml
-        - name: Upload coverage to Codecov
-          uses: codecov/codecov-action@v3
-          with:
-            file: ./coverage.xml
-            flags: unittests
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.12"]
+    runs-on: ${{ matrix.os }}
+    environment: openai1
+    steps:
+      # checkout to pr branch
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install packages and dependencies
+        run: |
+          docker --version
+          python -m pip install --upgrade pip wheel
+          pip install -e .[lmm]
+          python -c "import autogen"
+          pip install coverage pytest
+      - name: Coverage
+        env:
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        run: |
+          coverage run -a -m pytest test/agentchat/contrib/capabilities/test_image_generation_capability.py
+          coverage xml
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          file: ./coverage.xml
+          flags: unittests
+  TransformMessages:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-2019]
+        python-version: ["3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install packages and dependencies for all tests
+        run: |
+          python -m pip install --upgrade pip wheel
+          pip install pytest
+      - name: Install packages and dependencies for Transform Messages
+        run: |
+          pip install -e '.[long-context]'
+      - name: Set AUTOGEN_USE_DOCKER based on OS
+        shell: bash
+        run: |
+          if [[ ${{ matrix.os }} != ubuntu-latest ]]; then
+            echo "AUTOGEN_USE_DOCKER=False" >> $GITHUB_ENV
+          fi
+      - name: Coverage
+        run: |
+          pip install coverage>=5.3
+          coverage run -a -m pytest test/agentchat/contrib/capabilities/test_transform_messages.py --skip-openai
```

Comment:
Does this test require openai? If not, move to contrib-tests.yml

Reply:
I believe I accidentally removed the openai test from the workflow in the previous PR, however, the test is already implemented `test_transform_messages_capability`. `--skip-openai` was a copy-paste typo, removing it